src/file_operations.rs

```diff
@@ -86,6 +89,14 @@ impl FileOperations {
         self.operations.push(Op::MkDirAll(self.root.join(name)))
     }
 
+    pub fn link<P: AsRef<Path>>(&mut self, source: P, dest: P) {
```

Comment:
Isn't that what we want? I suppose you can imagine a case where there's a string and a path that get passed in but that seems like a prep step that should happen prior to calling this.